app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,35 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: %i(create destroy)
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build micropost_params
+    if @micropost.save
+      flash[:success] = t "controllers.microposts_controller.micropost_created"
+      redirect_to root_url
+    else
+      @feed_items = []
```

Comment:
tại sao phải khai báo @feed_items ở đây như thế này

Reply:
Khi gửi micropost không thành công . Trang chủ cần phải hủy những lần gửi thất bại đó và đơn giản nhất là gán cho nó 1 mảng trống sẽ triệt tiêu nguồn cấp dữ liệu hoàn toàn .

Reply:
hơi lủng củng, tìm hiểu lại nhé cái này nó liên quan tới render

Reply:
Trong app/view/shared/_feed.html.erb có sự dụng @feed_items.any? mà .any chỉ nhận giá trị array  , hash object nếu muốn thay [] bằng nil hoặc false thì có thể thay any bằng present .